packages/table/src/common/internal/scrollUtils.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { IRegion, RegionCardinality, Regions } from "../../regions";
+
+// Public
+// ======
+
+export function getScrollPositionForRegion(
+    region: IRegion,
+    currScrollLeft: number,
+    currScrollTop: number,
+    getLeftOffset: (columnIndex: number) => number,
+    getTopOffset: (rowIndex: number) => number,
+    numFrozenRows: number = 0,
+    numFrozenColumns: number = 0,
+) {
```

Comment:
A lot of params, but this way I don't have to pass a bloated `viewportRect` or `grid` object from `<Table>`. Thoughts?

Reply:
yeeesh. how many args would passing those props around save? this is an internal method and those args are passed by reference, so no perf penalty.

Reply:
It would save two args (`get{Top,Left}Offset => grid`, `currScroll{Top,Left} => viewportRect`), but the current args are much easier to mock for unit tests.

Also, we don't really _need_ to pass in the current scroll offsets. We could simply have this function return `null` for `scrollLeft` or `scrollTop` if that respective coordinate doesn't need to change, then the caller can substitute the current value for `null` in its scope. That reduces the params to:

```tsx
region: IRegion;
getLeftOffset: (columnIndex: number) => number;
getTopOffset: (rowIndex: number) => number;
numFrozenRows: number = 0;
numFrozenColumns: number = 0;
```